src/dotnet/Notification.Service/NewChatEntryNotificationGenerator.cs

```diff
@@ -0,0 +1,54 @@
+using ActualChat.Chat;
+using ActualChat.Chat.Events;
+
+namespace ActualChat.Notification;
+
+public class NewChatEntryNotificationGenerator: IChatEventNotificationGenerator<NewChatEntryEvent>
+{
+    private readonly IChatAuthorsBackend _chatAuthorsBackend;
+    private readonly IChatsBackend _chatsBackend;
+    private readonly MomentClockSet _clocks;
+
+    public NewChatEntryNotificationGenerator(
+        IChatAuthorsBackend chatAuthorsBackend,
+        IChatsBackend chatsBackend,
+        MomentClockSet clocks)
+    {
+        _chatAuthorsBackend = chatAuthorsBackend;
+        _chatsBackend = chatsBackend;
+        _clocks = clocks;
+    }
+
+    // TODO(AK): support mentions
+    // TODO(AK): throttle notifications
+    public async IAsyncEnumerable<NotificationEntry> GenerateNotifications(
```

Comment:
I would actually use a different API: generators should simply push the notifications to a common sink, + I suspect it is almost always a three-step process:

1. We identify users which may require a notif for a certain event
2. We trigger the generator for this user, which pushes a new notif
3. If there is a notification aggregator for this kind of notif, it kicks in & joins the last unseen notif w/ the same tag with the current one, marking the old one (unseen) obsolete.

Reply:
I prefer to postpone that to the moment when we start implementing inapp notifications